test/moq_messages.cpp

```diff
@@ -862,3 +862,75 @@ TEST_CASE("Subscribe Announces Error encode/decode")
     CHECK_EQ(msg.error_code, msg_out.error_code);
     CHECK_EQ(msg.reason_phrase, msg_out.reason_phrase);
 }
+
+static void
+FetchStreamEncodeDecode(bool extensions, bool empty_payload)
+{
+    Bytes buffer;
+    auto fetch_header = FetchHeader{};
+    fetch_header.subscribe_id = 0x1234;
+
+    buffer << fetch_header;
+
+    FetchHeader fetch_header_out;
+    CHECK(Verify(buffer, static_cast<uint64_t>(DataMessageType::kFetchHeader), fetch_header_out));
+    CHECK_EQ(fetch_header.subscribe_id, fetch_header_out.subscribe_id);
+
+    // stream all the objects
+    buffer.clear();
+    auto objects = std::vector<FetchObject>{};
+    // send 10 objects
+    for (size_t i = 0; i < 1; i++) {
+        auto obj = FetchObject{};
+        obj.group_id = 0x1234;
+        obj.subgroup_id = 0x5678;
+        obj.object_id = 0x9012;
+        obj.publisher_priority = 127;
+
+        if (empty_payload) {
+            obj.object_status = ObjectStatus::kDoesNotExist;
+        } else {
+            obj.payload = { 0x1, 0x2, 0x3, 0x4, 0x5 };
+        }
+
+        obj.extensions = extensions ? kOptionalExtensions : std::nullopt;
+        objects.push_back(obj);
+        buffer << obj;
+    }
+
+    auto obj_out = FetchObject{};
+    size_t object_count = 0;
+    StreamBuffer<uint8_t> in_buffer;
+    for (size_t i = 0; i < buffer.size(); i++) {
+        in_buffer.Push(buffer.at(i));
+        bool done;
+        done = in_buffer >> obj_out;
+        if (done) {
```

Comment:
Seems a bit odd.
```suggestion
        bool done = in_buffer >> obj_out;
        if (done) {
```
or even
```suggestion
        if (in_buffer >> obj_out) {
```

Reply:
I updated this, early continued, and matches the comment that said it was decoding (not 1) which seems maybe worthwhile.